### PR TITLE
Add doc about profiling Alloy

### PR DESCRIPTION
**Pull request by ptodev**
We often get asked to help with Alloy consumiing lots of memory. It would be helpful to point people to a doc. 

If there is a bug in Alloy that's causing the high consumption, then the doc would help people gather enough information to submit a GitHub issue. Even if there is no bug in Alloy, this doc could still help some users to gain more insight into what is consuming memory/CPU.

### CODE REVIEW

File: docs/sources/troubleshoot/profile.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+canonical: https://grafana.com/docs/alloy/latest/troubleshoot/profile/
+description: Learn how to profile resource consumption
+title: Profile Grafana Alloy resource consumption
+menuTitle: Profile resource consumption
+weight: 300
+---
+
+# Profile Grafana Alloy resource consumption
+
+Alloy is written in the Go programming language, which offers [built-in][pprof-pkg] support for profiling.
```
Comment (clayton-cornell) [inline docs/sources/troubleshoot/profile.md:11]:
```suggestion
{{< param "PRODUCT_NAME" >}} is written in the Go programming language, which offers [built-in][pprof-pkg] support for profiling.
```

File: docs/sources/troubleshoot/profile.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+Just like other applications written in Go, you can profile Alloy by sending an HTTP request which returns you a "pprof" Go profile.
```
Comment (clayton-cornell) [inline docs/sources/troubleshoot/profile.md:12]:
```suggestion
Like other applications written in Go, you can profile {{< param "PRODUCT_NAME" >}} by sending an HTTP request, which returns a pprof Go profile.
```

File: docs/sources/troubleshoot/profile.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+Once you have the pprof file, simply visualize it as a [flame graph][flame-graph] in [Grafana Pyroscope][pyroscope-getstarted].
+Alternatively, you could visualize it on [Grafana Play][pyroscope-adhoc] or by using Go's built-in [pprof tool][go-pprof] locally.
```
Comment (clayton-cornell) [inline docs/sources/troubleshoot/profile.md:15]:
```suggestion
Once you have the pprof file, visualize it as a [flame graph][flame-graph] in [Grafana Pyroscope][pyroscope-getstarted].
Alternatively, you could visualize it on [Grafana Play][pyroscope-adhoc] or locally by using Go's built-in [pprof tool][go-pprof].
```

File: docs/sources/troubleshoot/profile.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+[flame-graph]: https://grafana.com/docs/pyroscope/latest/view-and-analyze-profile-data/flamegraphs/
+[cmd-cli]: ../../reference/cli/run
+
+## Obtaining a single profile
+
+Different types of HTTP requests will retrieve different profiles.
+
+### Memory consumption
+
+Memory leaks are often caused by goroutine leaks.
```
Comment (clayton-cornell) [inline docs/sources/troubleshoot/profile.md:38]:
```suggestion
Goroutine leaks often cause memory leaks. 
```

Writing it this way is more active vs passive

File: docs/sources/troubleshoot/profile.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+Refer to the [Estimate resource usage][res-usage] page.
+
+[res-usage]: ../../introduction/estimate-resource-usage
+
+## What if my Alloy instance consumes an abnormally large amount of resources?
+
+You could open an issue in [Alloy's repository][alloy-repo]. 
+It would be helpful to attach pprof files and Alloy's configuration file.
+Make sure any secrets are redacted.
```
Comment (clayton-cornell) [inline docs/sources/troubleshoot/profile.md:99]:
```suggestion
If {{< param "PRODUCT_NAME" >}} consumes an abnormally large amount of resources, you can open an issue in the [Alloy repository][alloy-repo]. 
Attach your pprof files and your {{< param "PRODUCT_NAME" >}} configuration file.
Make sure you redact any secrets in the attachments.
```

File: docs/sources/troubleshoot/profile.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+[pyroscope components]: ../../reference/components/pyroscope
+
+## What is Alloy's expected resource consumption?
+
+Refer to the [Estimate resource usage][res-usage] page.
+
+[res-usage]: ../../introduction/estimate-resource-usage
+
+## What if my Alloy instance consumes an abnormally large amount of resources?
```
Comment (clayton-cornell) [inline docs/sources/troubleshoot/profile.md:95]:
```suggestion
## {{% param "PRODUCT_NAME" %}} consumes an abnormally large amount of resources
```

We should try to reduce the FAQ style headings

**Review by clayton-cornell — COMMENTED** [review]
First pass at an edit :-)

File: docs/sources/troubleshoot/profile.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+Once you have the pprof file, visualize it as a [flame graph][flame-graph] in [Grafana Pyroscope][pyroscope-getstarted].
+Alternatively, you could visualize it on [Grafana Play][pyroscope-adhoc] or locally by using Go's built-in [pprof tool][go-pprof].
+
+{{< admonition type="note" >}}
+A profile may contain sensitive information about your environment.
+You may not want to upload your profiles to a public location.
+{{< /admonition >}}
+
+The port you use to send the HTTP request is governed by Alloy's `--server.http.listen-addr` [command line argument][cmd-cli].
```
Comment (clayton-cornell) [inline docs/sources/troubleshoot/profile.md:22]:
```suggestion
The port you use to send the HTTP request is controlled by the `--server.http.listen-addr` [command line argument][cmd-cli].
```

Missed this one in the first pass.  I think we can drop the `Alloy's` and assume the reader knows we are talking about the CLI arg in Alloy here.

**Review by clayton-cornell — APPROVED** [review]
One minor tweak and we can get this one published

**Comment by clayton-cornell** [comment]
@ptodev Do we want to backport this? I am guessing yes?

**Comment by ptodev** [comment]
> @ptodev Do we want to backport this? I am guessing yes?

Sure, why not :) I added a backport label for all the previous releases. Thank you for the review!

**Pull request merged**